crates/boundless-market/src/contracts/mod.rs

```diff
@@ -523,6 +523,19 @@ impl Requirements {
     pub fn with_selector(self, selector: FixedBytes<4>) -> Self {
         Self { selector, ..self }
     }
+
+    /// Set the selector for an unaggregated proof.
+    ///
+    /// This will set the selector to the appropriate value based on the current environment.
+    /// In dev mode, the selector will be set to `FakeReceipt`, otherwise it will be set
+    /// to `Groth16V1_2`.
+    #[cfg(not(target_os = "zkvm"))]
+    pub fn with_unaggregated_proof(self) -> Self {
+        match risc0_zkvm::is_dev_mode() {
+            true => Self { selector: FixedBytes::from(Selector::FakeReceipt as u32), ..self },
+            false => Self { selector: FixedBytes::from(Selector::Groth16V1_2 as u32), ..self },
+        }
+    }
```

Comment:
Related to my other comment, but since we can initialize a mock verifier that expects a specific selector, is this necessary? It feels not ideal that we have to introduce the concept of dev mode into the client